message/observe.go

```diff
@@ -0,0 +1,33 @@
+// package message defines the terminology required for sharing
+// data between multiple PATH instances.
+package message
+
+// ObservationSet defines the functionality required for sharing
+// data between PATH instances.
+// e.g. QoS instances can share data by supplying an implementation
+// of this interface.
+type ObservationSet interface {
+	// MarshalJSON returns the serialized form
+	// of the set of observations, in JSON format.
+	// This is required for sharing QoS data between
+	// multiple PATH instances.
+	MarshalJSON() ([]byte, error)
+
+	// NotifyStakeHolders is used to communicate the observations contained
+	// in the set to the interested entities.
+	// e.g. The observation set returned by a gateway.ServiceRequestContext can
+	// guide the target service's QoS instance to update the quality data of 
+	// one or more endpoints.
+	NotifyStakeHolders() error
+}
+
+// TODO_FUTURE: consider using protobuf.
```

Comment:
I think this should be an up next if we go down this path. It'll be too much refactoring later.